### DISCUSSION TITLE
2021-09-22 Weekly Planning

### DISCUSSION THREAD

**Discussion by mathemancer** [mathesar-foundation/mathesar#677] [Maintainer discussions]
### Discussion topics

- **[Updates](https://github.com/centerofci/mathesar/discussions/677#discussioncomment-1368491)**: General updates everyone should be aware of.
- **[Questions](https://github.com/centerofci/mathesar/discussions/677#discussioncomment-1368528)**: Questions for the team to answer.
- **[Active Work](https://github.com/centerofci/mathesar/discussions/677#discussioncomment-1368537)**: Work planned for the next week.
- **[Retrospective](https://github.com/centerofci/mathesar/discussions/677#discussioncomment-1368541)**: Summary of last week's work.

### About

Each Tuesday, we start a discussion about Mathesar's progress. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

**Comment by mathemancer** [comment]
## Updates

### Progress
- [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4) is overdue.
- [06. 2021-09 Stability](https://github.com/centerofci/mathesar/milestone/34) is due 2021-09-23, and will likely be delayed.
- [07. Working with Tables](https://github.com/centerofci/mathesar/milestone/8) has only frontend issues remaining.
- [08. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) has both frontend and backend issues in progress.

We're currently experiencing delays, and will need to reorganize the roadmap once @kgodey returns.

**Comment by mathemancer** [comment]
## Questions

@powellc, @ghislaineguerin, @pavish: Your responses are encouraged. Others are also welcome to respond.

### Retrospective
- What went well over the past week?
- What could have gone better over the past week?
- Are there any improvements you'd like to suggest?

### Active Work
- Does your workload for this week look okay?
- We have some milestones with timing problems. Are there any suggestions for reprioritizing or reorganizing?  (Note that I don't expect to do this till @kgodey returns, but perhaps suggestions could help at that point.)

**Reply by pavish** [reply]
> What went well over the past week?

The issues for Milestone 5 are mostly complete. There is one open PR for adding new records which will be merged soon.

> What could have gone better over the past week?

I think the week went well, except that milestone 5 is overdue. 

> Are there any improvements you'd like to suggest?

None, at the moment.

> Does your workload for this week look okay?

The workload for this week is heavy. We will need reorganizing of milestones. I'm hoping to focus mostly on writing detailed tests, setting up integration test environment and to cover all scenarios for features that are complete.

> Are there any suggestions for reprioritizing or reorganizing?

None, at the moment.

**Reply by ghislaineguerin** [reply]
> What went well over the past week?

I was making good progress with the UI styling. I managed to have all view-related issues in review state. 

> What could have gone better over the past week?

I haven't made progress with the reviews. I think Kriti's input is needed.

> Are there any improvements you'd like to suggest?

I will change the review dates for the view-related issues to have Kriti as part of the process.

> Does your workload for this week look okay?

Yes

> We have some milestones with timing problems. Are there any suggestions for reprioritizing or reorganizing? (Note that I don't expect to do this till @kgodey returns, but perhaps suggestions could help at that point.)

None at the time

**Reply by mathemancer** [reply]
> * What went well over the past week?

Things are progressing on type implementations.  

> * What could have gone better over the past week?

The money types implementation has run into an area where most options seem pretty bad.
- Option 1:  too many types (one for each currency, about 300).  This means about 90,000 type casting functions, which is a problem.
- Option 2:  money currency is a display option.  These are not durable, and we could lose them under certain column manipulations
- Option 3:  money columns that save the currency in each row.  This requires setting the currency on each insert.  It's also not supported by the design.
The text types implementation ran into some bugs around server default setting.

> * Are there any improvements you'd like to suggest?

None

> * Does your workload for this week look okay?

I need to rework / reorganize.  I may have to put type implementation aside while server defaults are handled.

> * We have some milestones with timing problems. Are there any suggestions for reprioritizing or reorganizing?  (Note that I don't expect to do this till @kgodey returns, but perhaps suggestions could help at that point.)

No suggestions from me.

**Comment by mathemancer** [comment]
## Active Work

This post lists work planned for this week (see also: [Active](https://github.com/centerofci/mathesar/projects/1)). It also includes a current summary of milestones involved in this week's work.

### [2021-09-15] [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **[9 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2205.%20Editable%20Tables%22) already closed.** 
- **Issues not yet started:**
  - [@pavish] #197

### [2021-09-23] [06. 2021-09 Stability](https://github.com/centerofci/mathesar/milestone/34)

- **[4 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2206.%202021-09%20Stability%22) already closed.** 
- **Issues currently in progress:**
  - [@andyharley-cci] #225
  - [@ghislaineguerin] #473
- **Issues not yet started:**
  - [@kgodey] #592
  - [@kgodey] #280
  - [@kgodey] #623
  - [@kgodey] #583
  - [@ghislaineguerin] #636
  - [@ghislaineguerin] #638
  - [@pavish] #639
  - [@pavish] #640
  - [@pavish] #641
  - [@pavish] #642
  - [@pavish] #643
  - [@pavish] #213
- **Issues remaining in backlog:**
  - 5 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2206.%202021-09%20Stability%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2206.%202021-09%20Stability%22) issues
  - 3 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2206.%202021-09%20Stability%22) issues

### [2021-09-28] [07. Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **[20 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2207.%20Working%20with%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@mr-gabe49] #448
  - [@dmos62] #498
  - [@mr-gabe49] #496
  - [@mathemancer] #663
  - [@pegnott] #495
- **Issues remaining in backlog:**
  - 10 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2207.%20Working%20with%20Tables%22) issues

### [2021-09-30] [2021-09 improvements](https://github.com/centerofci/mathesar/milestone/30)

- **[6 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%222021-09%20improvements%22) already closed.** 
- **Issues currently in progress:**
  - [@arvind-1] #555
- **Issues remaining in backlog:**
  - 1 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%222021-09%20improvements%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%222021-09%20improvements%22) issues
  - 11 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%222021-09%20improvements%22) issues
  - 3 [infrastructure](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20infrastructure%22+milestone%3A%222021-09%20improvements%22) issues

### [2021-10-05] [08. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **[30 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2208.%20Initial%20Data%20Types%22) already closed.** 
- **Issues currently in progress:**
  - [@mathemancer] #417
  - [@eito-fis] #426
  - [@powellc] #385
  - [@powellc] #387
  - [@mathemancer] #407
  - [@mathemancer] #405
  - [@mathemancer] #404
- **Issues not yet started:**
  - [@powellc] #384
  - [@mathemancer] #412
  - [@mathemancer] #424
  - [@mathemancer] #430
  - [@mathemancer] #420
  - [@powellc] #410
- **Issues remaining in backlog:**
  - 17 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2208.%20Initial%20Data%20Types%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2208.%20Initial%20Data%20Types%22) issues
  - 9 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2208.%20Initial%20Data%20Types%22) issues
  - 8 [product](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20product%22+milestone%3A%2208.%20Initial%20Data%20Types%22) issues

### [2021-10-07] [09. Working with Views](https://github.com/centerofci/mathesar/milestone/13)

- **[4 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2209.%20Working%20with%20Views%22) already closed.** 
- **Issues currently in progress:**
  - [@ghislaineguerin] #455
  - [@ghislaineguerin] #443
  - [@ghislaineguerin] #466
  - [@ghislaineguerin] #454
  - [@ghislaineguerin] #243
  - [@ghislaineguerin] #456
- **Issues remaining in backlog:**
  - 7 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2209.%20Working%20with%20Views%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2209.%20Working%20with%20Views%22) issues
  - 2 [product](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20product%22+milestone%3A%2209.%20Working%20with%20Views%22) issues

### [2021-10-18] [11. Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **[7 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2211.%20Multiple%20Database%20Support%22) already closed.** 
- **Issues currently in progress:**
  - [@aaditkamat] #395
- **Issues remaining in backlog:**
  - 3 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2211.%20Multiple%20Database%20Support%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2211.%20Multiple%20Database%20Support%22) issues
  - 3 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2211.%20Multiple%20Database%20Support%22) issues
  - 2 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%2211.%20Multiple%20Database%20Support%22) issues

**Comment by mathemancer** [comment]
## Retrospective

This post lists issue updates from 2021-09-14 until now by milestone.

### [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **Issues closed:**
  - [@pavish] #195

### [06. 2021-09 Stability](https://github.com/centerofci/mathesar/milestone/34)

- **Issues closed:**
  - [@pavish] #651
- **Issues started, not yet finished:**
  - [@andyharley-cci] #225
  - [@ghislaineguerin] #473
- **New issues created:**
  - #668, created by @mr-gabe49

### [07. Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **Issues started, not yet finished:**
  - [@mr-gabe49] #448
  - [@dmos62] #498
  - [@mr-gabe49] #496
  - [@mathemancer] #663
  - [@pegnott] #495
- **New issues created:**
  - #663, created by @mr-gabe49

### [2021-09 improvements](https://github.com/centerofci/mathesar/milestone/30)

- **Issues started, not yet finished:**
  - [@arvind-1] #555

### [08. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **Issues closed:**
  - [@ghislaineguerin] #655
- **Issues started, not yet finished:**
  - [@mathemancer] #417
  - [@eito-fis] #426
  - [@powellc] #385
  - [@powellc] #387
  - [@mathemancer] #407
  - [@mathemancer] #405
  - [@mathemancer] #404
- **New issues created:**
  - #661, created by @pavish

### [09. Working with Views](https://github.com/centerofci/mathesar/milestone/13)

- **Issues started, not yet finished:**
  - [@ghislaineguerin] #455
  - [@ghislaineguerin] #443
  - [@ghislaineguerin] #466
  - [@ghislaineguerin] #454
  - [@ghislaineguerin] #243
  - [@ghislaineguerin] #456

### [11. Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **Issues started, not yet finished:**
  - [@aaditkamat] #395

### [2021-10 improvements](https://github.com/centerofci/mathesar/milestone/31)

- **New issues created:**
  - #670, created by @dmos62
